ibis/backends/tests/test_aggregation.py

```diff
@@ -572,23 +572,7 @@ def test_reduction_ops(
     reason="risingwave requires an `order_by` for these aggregations",
 )
 @pytest.mark.parametrize("method", ["first", "last"])
-@pytest.mark.parametrize(
-    "filtered",
-    [
-        param(
-            False,
-            marks=[
-                pytest.mark.notyet(
-                    ["datafusion"],
-                    raises=Exception,
-                    reason="datafusion 38.0.1 has a bug in FILTER handling that causes this test to fail",
-                    strict=False,
-                )
-            ],
-        ),
-        True,
-    ],
-)
```

Comment:
Most of these xfail removals are due to this upstream bug (https://github.com/apache/datafusion/issues/10878) being resolved. In some places we had a version-checked xfail, while in others we were always xfailing. Since there were multiple bad versions with the bug, I opted to do the easy thing and just remove all the xfails rather than messing with version checks. We don't do version checks for other non-local backends, so this seems fine to me.